### PR TITLE
HDDS-4199. Fix failed UT: TestOMAllocateBlockRequest#testValidateAndUpdateCache

### PR DESCRIPTION
**Pull request by runzhiwang**
## What changes were proposed in this pull request?

**What's the problem ?**

image

**What's the reason ?**

we can not make sure `omKeyInfo.getCreationTime()` different from ` omKeyInfo.getModificationTime()`

1. the creationTime was set in OmKeyInfo at [TestOMRequestUtils.addKeyToTable](https://github.com/apache/hadoop-ozone/blob/master/hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/key/TestOMAllocateBlockRequest.java#L68) -> [OmKeyInfo#createOmKeyInfo](https://github.com/apache/hadoop-ozone/blob/master/hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/TestOMRequestUtils.java#L253)
2. the modificationTime was set in KeyArgs at [doPreExecute(createAllocateBlockRequest())](https://github.com/apache/hadoop-ozone/blob/master/hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/key/TestOMAllocateBlockRequest.java#L58) -> [OMRequest#preExecute](https://github.com/apache/hadoop-ozone/blob/master/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMAllocateBlockRequest.java#L115). 
 And the KeyArgs.modificationTime was set in OmKeyInfo at [omAllocateBlockRequest.validateAndUpdateCache](https://github.com/apache/hadoop-ozone/blob/master/hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/key/TestOMAllocateBlockRequest.java#L84) -> [openKeyInfo.setModificationTime](https://github.com/apache/hadoop-ozone/blob/master/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMAllocateBlockRequest.java#L203).

3. Between [doPreExecute(createAllocateBlockRequest())](https://github.com/apache/hadoop-ozone/blob/master/hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/key/TestOMAllocateBlockRequest.java#L58) and [TestOMRequestUtils.addKeyToTable](https://github.com/apache/hadoop-ozone/blob/master/hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/key/TestOMAllocateBlockRequest.java#L68), there is no time consuming code, so we can not make sure the millisecond time is different.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-4199

## How was this patch tested?

no need new test.

### CODE REVIEW

**Comment by amaliujia** [comment]
Thanks for the investigation!

I am not sure if this is a good practice: will it help to insert a wait/sleep/a random operation after `doPreExecute(createAllocateBlockRequest());` to create the opportunity to differentiate  creation time and modification time

**Comment by runzhiwang** [comment]
@bharatviswa504 Could you help review this patch ? Thank you very much.

**Review by bharatviswa504 — COMMENTED** [review]
Instead of completely removing, can we check greater than or equal to?

**Comment by amaliujia** [comment]
In fact, greater than or equal to is a good idea!

File: hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/key/TestOMAllocateBlockRequest.java [hunk 1]
```diff
@@ -97,8 +95,11 @@ public void testValidateAndUpdateCache() throws Exception {
     // Check modification time
     Assert.assertEquals(modifiedOmRequest.getAllocateBlockRequest()
         .getKeyArgs().getModificationTime(), omKeyInfo.getModificationTime());
-    Assert.assertNotEquals(omKeyInfo.getCreationTime(),
-        omKeyInfo.getModificationTime());
+
+    // creationTime was assigned at TestOMRequestUtils.addKeyToTable
+    // modificationTime was assigned at doPreExecute(createAllocateBlockRequest())
+    Assert.assertTrue(
+        omKeyInfo.getCreationTime() <= omKeyInfo.getModificationTime());
```
Comment (amaliujia) [inline hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/key/TestOMAllocateBlockRequest.java:102]: +1 LGTM!

**Comment by runzhiwang** [comment]
@bharatviswa504 @amaliujia Thanks for review. I have updated the patch. Besides, I change the order of [doPreExecute(createAllocateBlockRequest())](https://github.com/apache/hadoop-ozone/blob/master/hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/key/TestOMAllocateBlockRequest.java#L58) and [TestOMRequestUtils.addKeyToTable](https://github.com/apache/hadoop-ozone/blob/master/hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/key/TestOMAllocateBlockRequest.java#L68), to make sure creationTime <= modificationTime, otherwise creationTime >= modificationTime looks weird.

**Comment by runzhiwang** [comment]
@bharatviswa504 CI passed. Could you help merge this patch ? Thank you very much.

**Review by adoroszlai — APPROVED** [review]
Thanks @runzhiwang for the fix.

**Pull request merged**